lib/api/objectCopy.js

```diff
@@ -91,10 +91,11 @@ function _prepMetadata(sourceObjMD, headers, sourceIsDestination, authInfo,
         });
         return { error: userMetadata };
     }
-    // If metadataDirective is 'COPY' and location constraint header is
-    // specified, new location constraint should override source object's
-    if (whichMetadata === 'COPY' && headers[locationHeader]) {
-        userMetadata[locationHeader] = headers[locationHeader];
+    // If metadataDirective is 'COPY' or 'REPLACE' with no location constraint,
+    // we use the bucket destination location constraint
+    if ((whichMetadata === 'COPY' || (whichMetadata === 'REPLACE' &&
```

Comment:
As discussed it doesn't make sense to me to copy the bucket location constraint to the obj location header if there was no obj location header on the original object. The check can be:
`if (sourceObjMd[locationHeader] && (whichMetadata === 'COPY' || (whichMetadata === 'REPLACE' && !userMetadata[locationHeader])))`
if that makes sense: if there was an obj location constraint on the original object and the directive is copy, or the directive is replace and there is no new obj location constraint, then we store the bucket location constraint as the obj location constraint